synfig-studio/src/gui/autorecover.cpp

```diff
@@ -164,8 +164,11 @@ AutoRecover::clear_backups()
 			// FileSystemTemporary will clear opened temporary files in destructor
 			String filename = App::get_temporary_directory() + ETL_DIRECTORY_SEPARATOR + *i;
 			bool s = false;
-			try { s = FileSystemTemporary("").open_temporary(filename); }
-			catch (...) { }
+			try {
+				FileSystemTemporary temporary_filesystem = FileSystemTemporary("");
+				s = temporary_filesystem.open_temporary(filename);
+				temporary_filesystem.discard_changes();
+			} catch (...) { }
```

Comment:
I think it is better to add synfig::warning on the exception. (Just to know what something bad is happened)